### DISCUSSION TITLE
Unable to compile scinterm

### DISCUSSION THREAD

**Discussion by eda2z** [orbitalquark/scinterm#5] [Q&A] [answered]
I am on the latest Arch linux and followed your instructions for building scinterm. All of the lexers compiled cleanly but when it got to ScintillaCurses.cxx, the following error occurred:

ScintillaCurses.cxx: In constructor ‘ScintillaCurses::ScintillaCurses(void (*)(void*, int, SCNotification*, void*), void*)’:
ScintillaCurses.cxx:973:31: error: invalid conversion from ‘void (*)(Scintilla::Surface*, Scintilla::PRectangle, int, const Scintilla::ViewStyle&)’ to ‘Scintilla::DrawTabArrowFn’ {aka ‘void (*)(Scintilla::Surface*, Scintilla::PRectangle, int)’} [-fpermissive]
  973 |     view.customDrawTabArrow = DrawTabArrow; // draw text arrows for tabs
      |                               ^~~~~~~~~~~~
      |                               |
      |                               void (*)(Scintilla::Surface*, Scintilla::PRectangle, int, const Scintilla::ViewStyle&)
make: *** [Makefile:35: ScintillaCurses.o] Error 1

I know next to nothing about C++. Any ideas? Thanks.

**Comment by orbitalquark** [comment, marked as answer]
The most recent revisions of Scinterm require a patch to Scintilla that was submitted upstream. Until the patch is accepted and there is a release that contains it, you can either apply this patch to Scintilla: https://github.com/orbitalquark/scinterm/issues/3#issuecomment-726195956, or you can use the most recent commit without the need for the patch: https://github.com/orbitalquark/scinterm/tree/1d14cbdb65435bcb64899d619b6299fb15b5130c
